core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -151,9 +151,21 @@ END_FUNC wait_secondary
 LOCAL_FUNC reset , :
 UNWIND(	.fnstart)
 UNWIND(	.cantunwind)
+
+/*
+ * if built-in arguments are configured, no arguments are expected from
+ * loader; instead, required parameters are configured in build time.
+ */
+
+#if defined (CFG_TEE_BUILT_IN_ARGS)
+    ldr r4, =CFG_TEE_PAGEABLE_ADDR
```

Comment:
Just want to avoid having to change the bootloader to pass information in R0, R2 and LR. There're existing firmware/loaders out there that are not aware of the OP-TEE's hand off protocol. Providing "built-in arguments" lowers the requirement as firmware/loader are only required "jump to" the OP-TEE Entry.
As to ARM Trusted Firmware, I think there is no need to deny the "built-in argument" when used together with it. For example, if the FW always passes the same arguments in each boot, there is no harm for OP-TEE to switch to use the built-in arguments.
As to ARMv8, I understand the concept of "built-in arguments" applies equally to ARMv8, however, for this change I do not planned to touch the ARMv8 code, because VxWorks does not support it for now and I do not have at hand any means to test the change.